File: mobile-android/app/src/main/java/com/erenlermarket/app/domain/repository/StorefrontRepository.kt
package com.erenlermarket.app.domain.repository

import com.erenlermarket.app.domain.model.Announcement
import com.erenlermarket.app.domain.model.StoreProfile

interface StorefrontRepository {
    suspend fun storeProfile(): StoreProfile
    suspend fun announcements(): List<Announcement>
}

File: mobile-ios/Modules/Domain/Sources/Repositories/StorefrontRepository.swift
/// Mağaza vitrini içeriği (profil + duyurular). Katalogdan ayrı tutulur.
public protocol StorefrontRepository: Sendable {
    func storeProfile(language: String) async throws -> StoreProfile
    func announcements(language: String) async throws -> [Announcement]
}
